pkg/chunked/storage_linux.go

```diff
@@ -230,21 +230,21 @@ func makeEstargzChunkedDiffer(ctx context.Context, store storage.Store, blobSize
 func copyFileFromOtherLayer(file *internal.FileMetadata, source string, otherFile *internal.FileMetadata, dirfd int, useHardLinks bool) (bool, *os.File, int64, error) {
 	srcDirfd, err := unix.Open(source, unix.O_RDONLY, 0)
 	if err != nil {
-		return false, nil, 0, err
+		return false, nil, 0, errors.Wrapf(err, "open source file %q", source)
 	}
 	defer unix.Close(srcDirfd)
 
 	srcFile, err := openFileUnderRoot(otherFile.Name, srcDirfd, unix.O_RDONLY, 0)
 	if err != nil {
-		return false, nil, 0, err
+		return false, nil, 0, errors.Wrapf(err, "open source file %q under target rootfs", otherFile.Name)
```

Comment:
I am worried about stutter here as well.  Doesn't this end up calling an unix.Open()? Which will report an error with the file that the open failed on?

Reply:
no, it won't use `unix.Open` but call directly the openat2 wrapper, that only produces the UNIX error without any additional information.